CSLabs.Api/Controllers/UserLabController.cs

```diff
@@ -46,6 +46,37 @@ public async Task<IActionResult> ProcessLastUsed()
             return Ok();
         }
 
+        [HttpGet("process-end-date-time")]
+        [AllowAnonymous]
+        public async Task<IActionResult> ProcessEndDateTime()
+        {
+            // find userLabs that have an EndDateTime paste the current UtcDateTime.
+            var userLabs = await DatabaseContext.UserLabs
+                .IncludeRelations()
+                .IncludeHypervisor()
+                .Include(l => l.UserLabVms)
+                .Where(l => l.EndDateTime > DateTime.UtcNow)
```

Comment:
Your query is the opposite of the check with the if statements,  You are checking if the EndDateTime is before now but then checking that it is after. In boolean logic this is called a Contradiction that ends up in the expression ending up with no items in the results.